packages/calcite-components/src/components/input-text/input-text.scss

```diff
@@ -1,4 +1,43 @@
+/**
+ * CSS Custom Properties
+ *
+ * These properties can be overridden using the component's tag as selector.
+ *
+ * @prop --calcite-input-text-corner-radius: defines the border radius of the component.
+ * @prop --calcite-input-text-text-color: defines the text color of the component.
+ * @prop --calcite-input-text-border-color: defines the border color of the component.
+ * @prop --calcite-input-text-background-color: defines the background color of the component.
+ * @prop --calcite-input-text-icon-color: defines the color of an icon element in the component.
+ * @prop --calcite-input-text-icon-color-hover: defines the color of an icon element when it's parent is hovered in the component.
+ * @prop --calcite-input-text-prefixsuffix-text-color: defines the text color of a prefix/suffix element in the component.
+ * @prop --calcite-input-text-prefixsuffix-background-color: defines the background color of a prefix/suffix element in the component.
+ * @prop --calcite-input-text-placeholder-text-color: defines the color of placeholder text in the component.
+ *
+ */
 :host {
+  --calcite-input-text-corner-radius: var(--calcite-border-radius-sharp);
```

Comment:
We'll want to conditionally apply this to the last rendered element - if the clear button is rendered it should be applied to the "outer" corners to prevent this state:

<img width="421" alt="Screenshot 2024-02-16 at 9 10 56 AM" src="https://github.com/Esri/calcite-design-system/assets/4733155/da17323c-b30b-4d05-84f6-1224e5d0eb1a">